File: app/src/main/java/com/applications/whazzup/photomapp/mvp/models/AbstractModel.java
package com.applications.whazzup.photomapp.mvp.models;


import com.applications.whazzup.photomapp.data.managers.DataManager;
import com.applications.whazzup.photomapp.data.managers.RealmManager;
import com.applications.whazzup.photomapp.di.components.DaggerModelComponent;
import com.applications.whazzup.photomapp.di.modules.ModelModule;

import javax.inject.Inject;

public class AbstractModel {

    @Inject
    DataManager mDataManager;

    @Inject
    RealmManager mRealmManager;

    public AbstractModel() {
        DaggerModelComponent.builder().modelModule(new ModelModule()).build().inject(this);
    }
}

File: app/src/main/java/com/applications/whazzup/photomapp/mvp/models/AbstractModel.kt
package com.applications.whazzup.photomapp.mvp.models


import com.applications.whazzup.photomapp.data.managers.DataManager
import com.applications.whazzup.photomapp.data.managers.RealmManager
import com.applications.whazzup.photomapp.di.components.DaggerModelComponent
import com.applications.whazzup.photomapp.di.modules.ModelModule
import com.birbit.android.jobqueue.JobManager
import javax.inject.Inject

abstract class AbstractModel {
    @Inject
    lateinit var mDataManager: DataManager

    @Inject
    lateinit var mRealmManager: RealmManager

    @Inject
    lateinit var mJobManger: JobManager

    init {
        DaggerModelComponent.builder().modelModule(ModelModule()).build().inject(this)
    }
}
